src/lsptoolshost/debugger/debugger.ts

```diff
@@ -51,9 +51,54 @@ export function registerDebugger(
     context.subscriptions.push(
         vscode.debug.registerDebugConfigurationProvider('coreclr', dotnetWorkspaceConfigurationProvider)
     );
+
     context.subscriptions.push(
-        vscode.commands.registerCommand('dotnet.generateAssets', async (selectedIndex) =>
-            generateAssets(workspaceInformationProvider, selectedIndex)
-        )
+        vscode.commands.registerCommand('dotnet.generateAssets', async (selectedIndex) => {
+            if (!(await promptForDevKitDebugConfigurations())) {
+                return;
+            }
+
+            await generateAssets(workspaceInformationProvider, selectedIndex);
+        })
     );
 }
+
+async function promptForDevKitDebugConfigurations(): Promise<boolean> {
+    if (getCSharpDevKit()) {
+        let result: boolean | undefined = undefined;
+
+        while (result === undefined) {
+            const labelYes = vscode.l10n.t('Yes');
+            const labelMoreInfo = vscode.l10n.t('More Information');
+            const title: string = vscode.l10n.t('.NET: Generate Assets for Build and Debug');
+
+            const dialogResult = await vscode.window.showInformationMessage(
+                title,
+                {
+                    modal: true,
+                    detail: vscode.l10n.t(
+                        `The '{0}' command is not reccomended to be used when C# Dev Kit extension is installed. Would you like build and debug using a dynamic configuration instead?`,
+                        title
+                    ),
+                },
+                labelYes,
+                labelMoreInfo
+            );
+
+            if (dialogResult === labelYes) {
+                await vscode.commands.executeCommand('workbench.action.debug.selectandstart', 'dotnet');
+                result = false;
+            } else if (dialogResult === labelMoreInfo) {
+                const launchjsonDescriptionURL = 'https://aka.ms/VSCode-CS-DynamicDebugConfig';
+                await vscode.env.openExternal(vscode.Uri.parse(launchjsonDescriptionURL));
+            } else if (dialogResult === undefined) {
+                // User cancelled dialog and wishes to continue generating assets.
+                result = true;
```

Comment:
Rather than doing this with a cancel button, can we add a `No` button that returns 'true' and either get rid of the 'Cancel' button, of if that isn't possible, have the 'Cancel' button return false? I feel like a button called 'Cancel' should cancel the command.

Reply:
Yeah, I'm not seeing a way ti hide "cancel" unless its non-modal but then we also lose the description.